### PR TITLE
Cdp etb integration

### PR DESCRIPTION
**Pull request by thanlanuot**

### CODE REVIEW

File: cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+require 'mechanize'
+require 'json'
+require 'time'
+
+module Integration
+  module CdpScraper
+    # The Base class for executing post/get as well as HTTP related tasks
+    # To be extended by EtbScraper and VtmScraper
+    #
+    class Base
```
Comment (tripledub) [inline cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb]: Base class should be in a separate file, so should children that inherit from it

File: cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+        check_key 'gender', params
+
+        check_valid 'gender', params, ['0', '1']
+        check_valid 'hiv_status', params, ['0', '1', '2', '3']
+
+        patient_list_page_url = uri('/#/quanlybenhnhan/sokhambenh')
+        get(patient_list_page_url)
+
+        params.merge!({
+          'ngayKhamBenh' => DateTime.strptime(params['diagnosis_date'], '%m/%d/%Y').strftime('%Y-%m-%dT00:00:00.000Z'),
```
Comment (tripledub) [inline cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb]: Did you see @jorgegorka s comment about date helpers?

File: cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+      def create_test_order(test_order)
+        params = test_order['test_order'].clone
+
+        post_data_get_patient = { "patientId": params['patient_vtm_id'] }
+        resp = post(uri('api/sokhambenh/GetPatient'), post_data_get_patient.to_json)
+        resp_json = JSON.parse(resp.body)
+        
+        vtm_id = resp_json['skbs'][0]['id']
+        
+        post_data = {
```
Comment (minhnghivn) [inline cdx_vietnam/app/middleware/middleware/integration/cdp_scraper.rb]: it is due to the nature of scraping job. For example, to obtain the patient-name from an HTML structure, we may have to: go to the second div of the first form[id=main] under the 3rd table tag...

File: cdx_vietnam/spec/models/integration_log_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
+
+describe IntegrationLog do
```
Comment (thanlanuot) [inline cdx_vietnam/spec/models/integration_log_spec.rb]: Currently, there is no validate in this model so there is no test

**Pull request merged**